website/docs/pipelines/promoting-applications.mdx

```diff
@@ -140,118 +142,167 @@ is upgraded in one of the pipeline's environments, a PR is created that updates
-3. Pipeline controller uses git provider api to create the pull request with the promoted manifests. A Personal Access Token (PAT) needs to be created to interact with pipelines git provider API. This PAT is also used to list pull requests from the configured repository.
+> It is also important to note that list and watch access also effectively allow for users to reveal the Secret contents.
 
-Create a Kubernetes secret with the previous data.
+3. Only allow creation of RBAC resources from paths where compliance controls are in place. For example, do not
+allow regular users creating RBAC resources using `Kubectl` or `Kubernetes API`.
 
-<details><summary>Expand to see example</summary>
+#### Policy
 
-```shell
-# example to use git over https with basic auth and pat
-$ kubectl create secret generic promotion-credentials \
-  --namespace=pipeline-01 \
-  --from-literal="username=<bot account name>" \
-  --from-literal="password=<token value>" \
-  --from-literal="token=<token value>" \
-  --from-literal="hmac-key=<hmac-key value>"
-```
+After configured RBAC, you have a static configuration. However,
+given there are no deny semantics in [RBAC](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#role-and-clusterrole).
+
+>An RBAC Role or ClusterRole contains rules that represent a set of permissions.
+> Permissions are purely additive (there are no "deny" rules).
+
+You should ensure that attempts to break this contract are blocked and detected. You could achieve it
+by leveraging [Policy capabilities](../../policy/intro/). Policy Agent acts in two complementary modes:
+- [Admission Controller](../../policy/intro#admission-controller) protects from any attempt to create non-compliant RBAC resources that would end granting access to the secret.
+- [Audit](../../policy/intro#audit) helps you identify already existing resources that are out of compliance. For example,
+roles created before policy agent was introduced as admission controller.
+
+Once enabled Policy, [Policy Library](../../policy/weave-policy-profile/) gives you have a set of policies to
+help you with the task. `RBACProhibitWildcards` and `RBACProhibitVerbsOnResources` are the building blocks
+that you could use to implement RBAC least privilege. Providing values to the policy parameters to restrict
+wildcards and access to secrets:
 
 ```yaml
----
-apiVersion: v1
-kind: Secret
-metadata:
-  name: promotion-credentials
-  namespace: pipeline-01
-data:
-  username: ZXhhbXBsZQ==
-  password: ZXhhbXBsZS1wYXNzd29yZA==
-  token: Z2hwX01IL3RsTFpXTXZMY0FxVWRYY1ZGL0lGbzh0WDdHNjdsZmRxWQ==
-  hmac-key: OEIzMTNBNjQ0REU0OEVGODgxMTJCQ0VFNTQ3NkE=
-type: Opaque
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+- policies/RBACProhibitWildcards/no-wildcard-on-resources-policy.yaml
```

Comment:
@MostafaMegahid how should we contribute these policies to the policy library ?

Reply:
Today we synced on this topic. the suggested path would be to 

1. add these policies to [policy library](https://github.com/weaveworks/policy-library/tree/main/policies). An example of this could https://github.com/weaveworks/policy-library/tree/main/policies/RBACClusterRoleClusterAdmin

Reply:
Draft PR for good practices within policy-library https://github.com/weaveworks/policy-library/pull/31

Reply:
To review today with Mostafa